packages/server/src/koaJsonRpc/index.ts

```diff
@@ -196,26 +166,17 @@ export default class KoaJsonRpc {
       return;
     }
 
-    // verify rate limit for batch request
-    const batchRequestTotalLimit = this.methodConfig[BATCH_REQUEST_METHOD_NAME].total;
-    // check rate limit for method and ip
-    if (this.rateLimit.shouldRateLimit(ctx.ip, BATCH_REQUEST_METHOD_NAME, batchRequestTotalLimit, this.requestId)) {
-      return jsonResp(null, new IPRateLimitExceeded(BATCH_REQUEST_METHOD_NAME), undefined);
-    }
-
     const response: any[] = [];
-    ctx.state.methodName = BATCH_REQUEST_METHOD_NAME;
 
     // we do the requests in parallel to save time, but we need to keep track of the order of the responses (since the id might be optional)
-    const promises = body.map((item: any) => {
+    const promises: Promise<any>[] = body.map(async (item: any) => {
+      ctx.state.methodName = item.method;
```

Comment:
Q: What was the lack of this line leading to? Empty string in some logs or low metrics numbers?
Good catch.

Reply:
Previously it was set to contain `BATCH_REQUEST_METHOD_NAME` for all requests inside the batch and now I have changed it to be the corresponding method name of each request that is being executed at that moment.

Reply:
Hmm, isn't there a single ctx for the batch method?
Will setting this as part of a map mean it's updated multiple times and the last set becomes its final value?